client-core/src/wallet/default_wallet_client.rs

```diff
@@ -105,161 +104,148 @@ where
     fn new_wallet(
         &self,
         name: &str,
-        passphrase: &SecUtf8,
+        enckey: &SecKey,
         wallet_kind: WalletKind,
     ) -> Result<Option<Mnemonic>> {
         #[cfg(not(debug_assertions))]
-        check_passphrase_strength(name, passphrase)?;
+        check_seckey_strength(name, enckey)?;
 
         match wallet_kind {
             WalletKind::Basic => {
                 let private_key = PrivateKey::new()?;
                 let view_key = PublicKey::from(&private_key);
 
                 self.key_service
-                    .add_keypair(&private_key, &view_key, passphrase)?;
+                    .add_keypair(&private_key, &view_key, enckey)?;
 
                 self.wallet_service
-                    .create(name, passphrase, view_key)
+                    .create(name, enckey, view_key)
                     .map(|_| None)
             }
             WalletKind::HD => {
                 let mnemonic = Mnemonic::new();
 
-                self.hd_key_service
-                    .add_mnemonic(name, &mnemonic, passphrase)?;
+                self.hd_key_service.add_mnemonic(name, &mnemonic, enckey)?;
 
-                let (public_key, private_key) = self.hd_key_service.generate_keypair(
-                    name,
-                    passphrase,
-                    HDAccountType::Viewkey,
-                )?;
+                let (public_key, private_key) =
+                    self.hd_key_service
+                        .generate_keypair(name, enckey, HDAccountType::Viewkey)?;
 
                 self.key_service
-                    .add_keypair(&private_key, &public_key, passphrase)?;
+                    .add_keypair(&private_key, &public_key, enckey)?;
 
-                self.wallet_service.create(name, passphrase, public_key)?;
+                self.wallet_service.create(name, enckey, public_key)?;
 
                 Ok(Some(mnemonic))
             }
         }
     }
 
-    fn restore_wallet(&self, name: &str, passphrase: &SecUtf8, mnemonic: &Mnemonic) -> Result<()> {
+    fn restore_wallet(&self, name: &str, enckey: &SecKey, mnemonic: &Mnemonic) -> Result<()> {
```

Comment:
I think `restore_wallet` should take `passphrase` and return HMAC. Also, add a function to get HMAC from `name` and `passphrase`.

Reply:
Done